### DISCUSSION TITLE
2fa, encryption, HTTPS

### DISCUSSION THREAD

**Discussion by velamvela** [drakkan/sftpgo#558] [Q&A] [answered]
1. How do i enable 2fa?
2. How do enable at rest encryption?
3. If i'm using sftp do i need to setup TLS certificates?   Are the TLS certificates only needed to secure the web login or is it needed for sftp users also for the best security?

**Comment by drakkan** [comment, marked as answer]
Hi,

2FA is available in the development version, you have to use edge based docker images or download a recent build artifact if you want to test it.

For data at rest, please take a look [here](https://github.com/drakkan/sftpgo/blob/main/docs/dare.md) to understand the basis and [here](https://github.com/drakkan/sftpgo/blob/main/docs/howto/getting-started.md#Creating-users-with-a-local-encrypted-backend-Data-At-Rest-Encryption) for an example configuration.

TLS certificates can be used for FTPS/FTPES/HTTPS, for SFTP you can use public key authentication

**Reply by velamvela** [reply]
Thank you my setup only uses username and password with sftp, so would it be safer to setup TLS?

**Reply by drakkan** [reply]
In this case TLS can be used to secure the web UI
